nautobot/core/api/metadata.py

```diff
@@ -256,11 +267,111 @@ def determine_metadata(self, request, view):
             metadata.update(
                 {
                     "schema": NautobotSchemaProcessor(serializer, request.parser_context).get_schema(),
-                    # "uiSchema": NautobotUiSchemaProcessor(serializer, request.parser_context).get_ui_schema(),
                     "uiSchema": ui_schema,
                 }
             )
 
             metadata["view_options"] = self.determine_view_options(request, serializer)
 
         return metadata
+
+    def restructure_view_config(self, view_config):
+        """
+        Restructure the view config by removing specific fields ("natural_key_slug", "url", "display", "status", "id")
+        from the view config and adding standard fields ("id", "natural_key_slug", "url") to the first item's fields.
+
+        This operation aims to establish a standardized and consistent way of displaying the fields "id", "natural_key_slug",
+        and "url" within the view config.
+
+        Example:
+            >>> view_config = [
+                {
+                    Location: {fields: ["display", "name",...]},
+                    Others: {fields: ["tenant", "tenant_group", "status"]}
+                },
+                ...
+            ]
+            >>> restructure_view_config(view_config)
+            [
+                {
+                    Location: {fields: ["name","id","natural_key_slug","url"]},
+                    Others: {fields: ["tenant", "tenant_group"]}
+                },
+                ...
+            ]
+        """
+
+        # TODO(timizuo): Add a standardized way of handling `tenant` and `tags` fields, Possible should be on last items on second col.
+        fields_to_remove = ["natural_key_slug", "url", "display", "status", "id"]
+        fields_to_add = ["id", "natural_key_slug", "url"]
```

Comment:
This will of course conflict with #3881.

Reply:
Its now `composite_key`?

Reply:
Yes.